src/components/SpectralProfiler/ProfileFittingComponent/ProfileFittingComponent.tsx

```diff
@@ -0,0 +1,443 @@
+            // if (this.props.widgetStore.smoothingStore.type !== SmoothingType.NONE) {
+            //     const smoothingData = this.props.widgetStore.smoothingStore.getSmoothingValues(fittingData.rawX, fittingData.rawY);
+            //     this.props.fittingStore.autoDetect(smoothingData.x, Array.prototype.slice.call(smoothingData.y));
+            // } else {
+                this.props.fittingStore.autoDetect();
+            // }
+            if (this.props.fittingStore.isAutoDetectWithFitting) {
+                this.fitData();
+            }
+        }
+        this.props.fittingStore.setHasAutoDetectResult(true);
+    }
+
+    private deleteComponent = () => {
+        this.props.fittingStore.deleteSelectedComponent();
+    }
+
+    private cursorSelecting = () => {
+        this.props.fittingStore.setIsCursorSelectionOn(!this.props.fittingStore.isCursorSelectionOn);
+    }
+
+    private onCenterLocked = () => {
+        this.props.fittingStore.selectedComponent.setLockedCenter(!this.props.fittingStore.selectedComponent.lockedCenter);
+    }
+
+    private onAmpLocked = () => {
+        this.props.fittingStore.selectedComponent.setLockedAmp(!this.props.fittingStore.selectedComponent.lockedAmp);
+    }
+
+    private onFwhmLocked = () => {
+        this.props.fittingStore.selectedComponent.setLockedFwhm(!this.props.fittingStore.selectedComponent.lockedFwhm);
+    }
+
+    private showLog = () => {
+        this.setIsShowingLog(true);
+    }
+
+    private handleLogClose = () => {
+        this.setIsShowingLog(false);
+    }
+
+    private saveLog = () => {
+        let headerString = "";
+        const frame = this.props.widgetStore.effectiveFrame;
+        if (frame && frame.frameInfo && frame.regionSet) {
+            headerString += `# image: ${frame.filename}\n`;
+
+            const regionId = this.props.widgetStore.effectiveRegionId;
+            const region = frame.regionSet.regions.find(r => r.regionId === regionId);
+
+            // statistic type, ignore when region == cursor
+            if (regionId !== 0) {
+                headerString += `# statistic: ${this.props.widgetStore.profileSelectionStore.selectedStatsTypes[0]}\n`;
+            }
+            // region info
+            if (region) {
+                headerString += `# ${region.regionProperties}\n`;
+                if (frame.validWcs) {
+                    headerString += `# ${frame.getRegionWcsProperties(region)}\n`;
+                }
+            }
+        }
+
+        const content = `data:text/plain;charset=utf-8,${headerString}\n${this.props.fittingStore.resultLog}\n`;
+        const dataURL = encodeURI(content).replace(/#/g, "%23");
+
+        const a = document.createElement("a") as HTMLAnchorElement;
+        a.href = dataURL;
+
+        a.download = `Profile Fitting Result Log-${getTimestamp()}.txt`;
+        a.dispatchEvent(new MouseEvent("click"));
+    }
+
+    private reset = () => {
```

Comment:
I think perhaps this should also be an `@action`, so that all these changes happen at once without side effects

Reply:
modified.